### PR TITLE
Document the behaviour difference between LLD and GNU LD while using --defsym for library selection

### PR DESCRIPTION
**Pull request by simpal01**

### CODE REVIEW

File: docs/migrating.md [hunk 1]
```diff
@@ -100,6 +100,9 @@ however there are some known differences to take into account, see:
 in the LLD documentation.
 * The [LLD and GNU linker incompatibilities](https://maskray.me/blog/2020-12-19-lld-and-gnu-linker-incompatibilities)
 blog post.
+* When selecting the printf/scanf library variant with `--defsym from=to`, lld includes not only the specified
```
Comment (voltur01) [inline docs/migrating.md:103]: capitalize LLD as above.

Comment (voltur01) [reply docs/migrating.md:103]: "library variant" - implementation variant?
and maybe further add implementation variant from picolibc

Comment (voltur01) [inline docs/migrating.md:103]: add a reference to https://github.com/picolibc/picolibc/blob/main/doc/printf.md after the defsym to be clear when/how it is used.

File: docs/migrating.md [hunk 2]
```diff
@@ -100,6 +100,9 @@ however there are some known differences to take into account, see:
...
+printf/scanf library variant through defsym but also the PICOLIBC_DOUBLE_PRINTF_SCANF variant. To avoid linking
```
Comment (voltur01) [inline docs/migrating.md:104]: do you want to add code formatting for defsym? or ever refer to it as `--defsym`?

File: docs/migrating.md [hunk 3]
```diff
@@ -100,6 +100,9 @@ however there are some known differences to take into account, see:
...
+them into the application, you must include the --gc-sections linker flag.
```
Comment (voltur01) [inline docs/migrating.md:105]: "them" maybe "the default variant"?

Comment (voltur01) [inline docs/migrating.md:105]: code format --gc-sections?

**Review by voltur01 — COMMENTED** [review]
Thank you for adding this! Please see some style and reference suggestions.

**Review by voltur01 — COMMENTED** [review]
Thank you for the update! Please see some minor suggestions for your consideration.

**Comment by nasherm** [comment]
LGTM

**Review by nasherm — APPROVED** [review]
LGTM

**Pull request merged**